vcsutils/utils_test.go

```diff
@@ -97,11 +96,40 @@ func TestDiscardResponseBody(t *testing.T) {
 }
 
 func openTarball(t *testing.T) (string, *os.File) {
-	dir, err := ioutil.TempDir("", "")
+	dir, err := os.MkdirTemp("", "")
 	assert.NoError(t, err)
 	defer os.RemoveAll(dir)
 
 	tarball, err := os.Open(filepath.Join("testdata", "a.tar.gz"))
 	assert.NoError(t, err)
 	return dir, tarball
 }
+
+func TestDefaultIfNotNil(t *testing.T) {
+	str := "Hello world"
+	assert.Equal(t, "Hello world", DefaultIfNotNil(&str))
+	assert.Equal(t, "", DefaultIfNotNil(new(string)))
+	num := 3
+	assert.Equal(t, 3, DefaultIfNotNil(&num))
+	assert.Equal(t, 0, DefaultIfNotNil(new(int)))
+	boolVal := true
+	assert.Equal(t, true, DefaultIfNotNil(&boolVal))
+	assert.Equal(t, false, DefaultIfNotNil(new(bool)))
+	arr := []int{3, 4}
+	assert.Equal(t, []int{3, 4}, DefaultIfNotNil(&arr))
+	assert.Equal(t, []int(nil), DefaultIfNotNil(new([]int)))
+}
+
+func TestUnzip(t *testing.T) {
+	destDir, err := os.MkdirTemp("", "")
+	assert.NoError(t, err)
+	defer os.RemoveAll(destDir)
```

Comment:
We should assert the error returned by the `RemoveAll` function. There's another usage usage of `RemoveAll` in this function that requires this fix as well.

Reply:
done